### ISSUE TITLE
"Cannot read properties of undefined (reading 't')" from ESM changes in 12.1

### ISSUE THREAD

**Issue by joshkel** [i18next/react-i18next#1595]
## 💥 Regression Report

Adding the `exports` section in 12.1.0 resulted in a bug for our application:

https://github.com/i18next/react-i18next/blob/v12.1.1/package.json#L20

The problem was caused by how we use i18next: We have an in-house NPM library package that handles common internationalization tasks such as initializing react-i18next, then one or more packages that use Webpack to create web apps. Our in-house libraries still use CommonJS, while our Babel + Webpack configuration apparently supports ESM.

As a result, we were pulling in two different versions of react-i18next (`react-i18next/dist/es/index.js` and `react-i18next/dist/commonjs/index.js`) and only initializing one.

## Last working version

Worked up to version: 12.0.0

Stopped working in version: 12.1.1

## To Reproduce

Access react-i18next in two fashions, one which supports ESM and one which doesn't. Accessing the uninitialized i18next instance then results in errors similar to

```
TypeError: Cannot read properties of undefined (reading 't')
  at Trans(/static/js/bundle.js:383638:30)
```

## Expected behavior

react-i18next works - or, at least, changes to bundler and module usage are called out in the changelog.

## Your Environment

- *runtime version*: node v14 plus browser with Webpack
- *i18next version*: 22.4.5
- *os*: Mac

After investigation, this seems like a misconfiguration (or, at least, an unsupported configuration) on my part, so feel free to close this issue without any action if appropriate. I at least wanted to document it in case it helps anyone. Thank you.

**Comment by adrai** [comment]
Never seen such a setup, if you like, please provide more information and also how you "fixed" it.

**Comment by joshkel** [comment]
Thanks for the follow-up.

More broadly, I believe this scenario could happen whenever you have the following:

* Package `A` uses ESM and depends on react-i18next
* Package `B` uses CommonJS and depends on react-i18next
* Package `A` depends on package `B` (or, perhaps, vice versa - I haven't investigated)

The "fixes" that I've used aren't great - downgrading to react-i18next 12.0.0 works, and I've also used a [Yarn patch](https://yarnpkg.com/cli/patch) to remove the `exports` section from react-i18next 12.1.x's package.json. The long-term fix is to get everything on ESM, but our codebase and its dependencies aren't there yet.
